src/foam/u2/grid/GridHeaderCell.js

```diff
@@ -0,0 +1,146 @@
+foam.CLASS
+({
+    package: "foam.u2.grid",
+    name: "GridHeaderCell",
+    extends: "foam.u2.Element",
+
+    imports: [
+    ], 
+    
+    axioms: [
+      foam.u2.CSS.create({
+          code: function() {/*
+                ^grid-header-cell {
+                    border: 2px solid black;
+                }
+                ^cell-header {
+                    line-height: 100%;
+                    font-size: 120%;
+                    font-weight: bold;
+                    text-align: center;
+                }
+                ^row-header-highlight {
+                    background-color:rgba(0, 0, 255, 0.3);
+                }
+                ^col-header-highlight {
+                    background-color:rgba(0, 255, 0, 0.3);
+                }
+          */}
+        })
+      ],
+    
+    properties: [
+        'of',
+        {
+            name: "cell",
+            factory: function(){
+                var b  = foam.u2.Element.create();
+                b.setNodeName("div");
+                return b; 
+            }
+        }, 
+        'property',
+        'data', 
+        //rowIndex and colIndex keeps track of if it's row or col header, and which row/col it is. 
+        {
+            name: 'rowIndex',
+            value: -1, 
+        },
+        {
+            name: 'colIndex',
+            value: -1, 
+        }, 
+        {
+            name: 'name',
+            expression: function(data){
+                if (typeof data == "string"){
+                    return data; 
+                }else if (data && data.name){
+                    return data.name; 
+                }
+                return undefined; 
+            }
+        },
+        {
+            name: "M",
+            factory: function() { return foam.mlang.Expressions.create();},
+        }, 
+        
+        {
+            name: 'isSelected',
+            value: false, 
+        }, 
+        {
+            name: 'headerHighlightCSSClass',
+            expression: function(isSelected, rowIndex, colIndex){
+                if (isSelected){
+                    if (rowIndex >-1){
+                        return this.myCls("row-header-highlight");
+                    }else if (colIndex >-1){
+                        return this.myCls("col-header-highlight");
+                    }
+                }
+            }
+        },
+
+      ], 
+    methods: [
+        function initE() {
+            this.cell.cssClass(this.myCls("cell-header"));
```

Comment:
https://en.wikipedia.org/wiki/Fluent_interface

Basically, a method which returns 'this' so that method calls can be chained.
So instead of:
this.a();
this.b();
this.c();
You can just do this.a().b().c();

Many of the u2.Element methods are designed this way. If you have at the FOAM u2 code or demo code, you'll see this style employed. It gives a more HTML-like structure to the code and makes it easier to follow.